### ISSUE TITLE
依照detection.md上的步骤尝试文字检测模型的训练，不能保存模型

### ISSUE THREAD

**Issue by Sest0** [PaddlePaddle/PaddleOCR#2142]
[2021/03/01 11:55:55] root INFO: Architecture : 
[2021/03/01 11:55:55] root INFO:     Backbone : 
[2021/03/01 11:55:55] root INFO:         model_name : large
[2021/03/01 11:55:55] root INFO:         name : MobileNetV3
[2021/03/01 11:55:55] root INFO:         scale : 0.5
[2021/03/01 11:55:55] root INFO:     Head : 
[2021/03/01 11:55:55] root INFO:         k : 50
[2021/03/01 11:55:55] root INFO:         name : DBHead
[2021/03/01 11:55:55] root INFO:     Neck : 
[2021/03/01 11:55:55] root INFO:         name : DBFPN
[2021/03/01 11:55:55] root INFO:         out_channels : 256
[2021/03/01 11:55:55] root INFO:     Transform : None
[2021/03/01 11:55:55] root INFO:     algorithm : DB
[2021/03/01 11:55:55] root INFO:     model_type : det
[2021/03/01 11:55:55] root INFO: Eval : 
[2021/03/01 11:55:55] root INFO:     dataset : 
[2021/03/01 11:55:55] root INFO:         data_dir : ./train_data/icdar2015/text_localization/
[2021/03/01 11:55:55] root INFO:         label_file_list : ['./train_data/icdar2015/text_localization/test_icdar2015_label.txt']
[2021/03/01 11:55:55] root INFO:         name : SimpleDataSet
[2021/03/01 11:55:55] root INFO:         transforms : 
[2021/03/01 11:55:55] root INFO:             DecodeImage : 
[2021/03/01 11:55:55] root INFO:                 channel_first : False
[2021/03/01 11:55:55] root INFO:                 img_mode : BGR
[2021/03/01 11:55:55] root INFO:             DetLabelEncode : None
[2021/03/01 11:55:55] root INFO:             DetResizeForTest : 
[2021/03/01 11:55:55] root INFO:                 image_shape : [736, 1280]
[2021/03/01 11:55:55] root INFO:             NormalizeImage : 
[2021/03/01 11:55:55] root INFO:                 mean : [0.485, 0.456, 0.406]
[2021/03/01 11:55:55] root INFO:                 order : hwc
[2021/03/01 11:55:55] root INFO:                 scale : 1./255.
[2021/03/01 11:55:55] root INFO:                 std : [0.229, 0.224, 0.225]
[2021/03/01 11:55:55] root INFO:             ToCHWImage : None
[2021/03/01 11:55:55] root INFO:             KeepKeys : 
[2021/03/01 11:55:55] root INFO:                 keep_keys : ['image', 'shape', 'polys', 'ignore_tags']
[2021/03/01 11:55:55] root INFO:     loader : 
[2021/03/01 11:55:55] root INFO:         batch_size_per_card : 1
[2021/03/01 11:55:55] root INFO:         drop_last : False
[2021/03/01 11:55:55] root INFO:         num_workers : 8
[2021/03/01 11:55:55] root INFO:         shuffle : False
[2021/03/01 11:55:55] root INFO:         use_shared_memory : False
[2021/03/01 11:55:55] root INFO: Global : 
[2021/03/01 11:55:55] root INFO:     cal_metric_during_train : False
[2021/03/01 11:55:55] root INFO:     checkpoints : None
[2021/03/01 11:55:55] root INFO:     debug : False
[2021/03/01 11:55:55] root INFO:     distributed : False
[2021/03/01 11:55:55] root INFO:     epoch_num : 1200
[2021/03/01 11:55:55] root INFO:     eval_batch_step : [0, 10]
[2021/03/01 11:55:55] root INFO:     infer_img : doc/imgs_en/img_10.jpg
[2021/03/01 11:55:55] root INFO:     load_static_weights : True
[2021/03/01 11:55:55] root INFO:     log_smooth_window : 20
[2021/03/01 11:55:55] root INFO:     pretrained_model : ./pretrain_models/MobileNetV3_large_x0_5_pretrained
[2021/03/01 11:55:55] root INFO:     print_batch_step : 10
[2021/03/01 11:55:55] root INFO:     save_epoch_step : 200
[2021/03/01 11:55:55] root INFO:     save_inference_dir : None
[2021/03/01 11:55:55] root INFO:     save_model_dir : ./output/db_mv3/
[2021/03/01 11:55:55] root INFO:     save_res_path : ./output/det_db/predicts_db.txt
[2021/03/01 11:55:55] root INFO:     use_gpu : True
[2021/03/01 11:55:55] root INFO:     use_visualdl : False
[2021/03/01 11:55:55] root INFO: Loss : 
[2021/03/01 11:55:55] root INFO:     alpha : 5
[2021/03/01 11:55:55] root INFO:     balance_loss : True
[2021/03/01 11:55:55] root INFO:     beta : 10
[2021/03/01 11:55:55] root INFO:     main_loss_type : DiceLoss
[2021/03/01 11:55:55] root INFO:     name : DBLoss
[2021/03/01 11:55:55] root INFO:     ohem_ratio : 3
[2021/03/01 11:55:55] root INFO: Metric : 
[2021/03/01 11:55:55] root INFO:     main_indicator : hmean
[2021/03/01 11:55:55] root INFO:     name : DetMetric
[2021/03/01 11:55:55] root INFO: Optimizer : 
[2021/03/01 11:55:55] root INFO:     base_lr : 0.0001
[2021/03/01 11:55:55] root INFO:     beta1 : 0.9
[2021/03/01 11:55:55] root INFO:     beta2 : 0.999
[2021/03/01 11:55:55] root INFO:     lr : 
[2021/03/01 11:55:55] root INFO:         learning_rate : 0.001
[2021/03/01 11:55:55] root INFO:     name : Adam
[2021/03/01 11:55:55] root INFO:     regularizer : 
[2021/03/01 11:55:55] root INFO:         factor : 0
[2021/03/01 11:55:55] root INFO:         name : L2
[2021/03/01 11:55:55] root INFO: PostProcess : 
[2021/03/01 11:55:55] root INFO:     box_thresh : 0.6
[2021/03/01 11:55:55] root INFO:     max_candidates : 1000
[2021/03/01 11:55:55] root INFO:     name : DBPostProcess
[2021/03/01 11:55:55] root INFO:     thresh : 0.3
[2021/03/01 11:55:55] root INFO:     unclip_ratio : 1.5
[2021/03/01 11:55:55] root INFO: Train : 
[2021/03/01 11:55:55] root INFO:     dataset : 
[2021/03/01 11:55:55] root INFO:         data_dir : ./train_data/icdar2015/text_localization/
[2021/03/01 11:55:55] root INFO:         label_file_list : ['./train_data/icdar2015/text_localization/train_icdar2015_label.txt']
[2021/03/01 11:55:55] root INFO:         name : SimpleDataSet
[2021/03/01 11:55:55] root INFO:         ratio_list : [1.0]
[2021/03/01 11:55:55] root INFO:         transforms : 
[2021/03/01 11:55:55] root INFO:             DecodeImage : 
[2021/03/01 11:55:55] root INFO:                 channel_first : False
[2021/03/01 11:55:55] root INFO:                 img_mode : BGR
[2021/03/01 11:55:55] root INFO:             DetLabelEncode : None
[2021/03/01 11:55:55] root INFO:             IaaAugment : 
[2021/03/01 11:55:55] root INFO:                 augmenter_args : 
[2021/03/01 11:55:55] root INFO:                     args : 
[2021/03/01 11:55:55] root INFO:                         p : 0.5
[2021/03/01 11:55:55] root INFO:                     type : Fliplr
[2021/03/01 11:55:55] root INFO:                     args : 
[2021/03/01 11:55:55] root INFO:                         rotate : [-10, 10]
[2021/03/01 11:55:55] root INFO:                     type : Affine
[2021/03/01 11:55:55] root INFO:                     args : 
[2021/03/01 11:55:55] root INFO:                         size : [0.5, 3]
[2021/03/01 11:55:55] root INFO:                     type : Resize
[2021/03/01 11:55:55] root INFO:             EastRandomCropData : 
[2021/03/01 11:55:55] root INFO:                 keep_ratio : True
[2021/03/01 11:55:55] root INFO:                 max_tries : 50
[2021/03/01 11:55:55] root INFO:                 size : [640, 640]
[2021/03/01 11:55:55] root INFO:             MakeBorderMap : 
[2021/03/01 11:55:55] root INFO:                 shrink_ratio : 0.4
[2021/03/01 11:55:55] root INFO:                 thresh_max : 0.7
[2021/03/01 11:55:55] root INFO:                 thresh_min : 0.3
[2021/03/01 11:55:55] root INFO:             MakeShrinkMap : 
[2021/03/01 11:55:55] root INFO:                 min_text_size : 8
[2021/03/01 11:55:55] root INFO:                 shrink_ratio : 0.4
[2021/03/01 11:55:55] root INFO:             NormalizeImage : 
[2021/03/01 11:55:55] root INFO:                 mean : [0.485, 0.456, 0.406]
[2021/03/01 11:55:55] root INFO:                 order : hwc
[2021/03/01 11:55:55] root INFO:                 scale : 1./255.
[2021/03/01 11:55:55] root INFO:                 std : [0.229, 0.224, 0.225]
[2021/03/01 11:55:55] root INFO:             ToCHWImage : None
[2021/03/01 11:55:55] root INFO:             KeepKeys : 
[2021/03/01 11:55:55] root INFO:                 keep_keys : ['image', 'threshold_map', 'threshold_mask', 'shrink_map', 'shrink_mask']
[2021/03/01 11:55:55] root INFO:     loader : 
[2021/03/01 11:55:55] root INFO:         batch_size_per_card : 16
[2021/03/01 11:55:55] root INFO:         drop_last : False
[2021/03/01 11:55:55] root INFO:         num_workers : 8
[2021/03/01 11:55:55] root INFO:         shuffle : True
[2021/03/01 11:55:55] root INFO:         use_shared_memory : False
[2021/03/01 11:55:55] root INFO: train with paddle 2.0.0 and device CUDAPlace(0)
[2021/03/01 11:55:55] root INFO: Initialize indexs of datasets:['./train_data/icdar2015/text_localization/train_icdar2015_label.txt']
[2021/03/01 11:55:55] root INFO: Initialize indexs of datasets:['./train_data/icdar2015/text_localization/test_icdar2015_label.txt']
[2021/03/01 11:55:59] root INFO: load pretrained model from ['./pretrain_models/MobileNetV3_large_x0_5_pretrained']
[2021/03/01 11:55:59] root INFO: train dataloader has 63 iters, valid dataloader has 500 iters
[2021/03/01 11:55:59] root INFO: During the training process, after the 0th iteration, an evaluation is run every 10 iterations
[2021/03/01 11:55:59] root INFO: Initialize indexs of datasets:['./train_data/icdar2015/text_localization/train_icdar2015_label.txt']

每次运行到这里过一段时间就会停住，不保存模型。
新人上路，还请各位指导一下，谢谢

**Comment by Sest0** [comment]
det_mv3_db.yml文件的设置
Global:
  use_gpu: true
  epoch_num: 1200
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/db_mv3/
  save_epoch_step: 200
  # evaluation is run every 2000 iterations
  eval_batch_step: [0, 10] #[0,2000]
  # if pretrained_model is saved in static mode, load_static_weights must set to True
  load_static_weights: True
  cal_metric_during_train: False
  pretrained_model: ./pretrain_models/MobileNetV3_large_x0_5_pretrained
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img: doc/imgs_en/img_10.jpg
  save_res_path: ./output/det_db/predicts_db.txt

Architecture:
  model_type: det
  algorithm: DB
  Transform:
  Backbone:
    name: MobileNetV3
    scale: 0.5
    model_name: large
  Neck:
    name: DBFPN
    out_channels: 256
  Head:
    name: DBHead
    k: 50

Loss:
  name: DBLoss
  balance_loss: true
  main_loss_type: DiceLoss
  alpha: 5
  beta: 10
  ohem_ratio: 3

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    learning_rate: 0.001
  regularizer:
    name: 'L2'
    factor: 0

PostProcess:
  name: DBPostProcess
  thresh: 0.3
  box_thresh: 0.6
  max_candidates: 1000
  unclip_ratio: 1.5

Metric:
  name: DetMetric
  main_indicator: hmean

Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/icdar2015/text_localization/
    label_file_list:
      - ./train_data/icdar2015/text_localization/train_icdar2015_label.txt
    ratio_list: [1.0]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - DetLabelEncode: # Class handling label
      - IaaAugment:
          augmenter_args:
            - { 'type': Fliplr, 'args': { 'p': 0.5 } }
            - { 'type': Affine, 'args': { 'rotate': [-10, 10] } }
            - { 'type': Resize, 'args': { 'size': [0.5, 3] } }
      - EastRandomCropData:
          size: [640, 640]
          max_tries: 50
          keep_ratio: true
      - MakeBorderMap:
          shrink_ratio: 0.4
          thresh_min: 0.3
          thresh_max: 0.7
      - MakeShrinkMap:
          shrink_ratio: 0.4
          min_text_size: 8
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: ['image', 'threshold_map', 'threshold_mask', 'shrink_map', 'shrink_mask'] # the order of the dataloader list
  loader:
    shuffle: True
    drop_last: False
    batch_size_per_card: 16
    num_workers: 8
    use_shared_memory: False

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/icdar2015/text_localization/
    label_file_list:
      - ./train_data/icdar2015/text_localization/test_icdar2015_label.txt
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - DetLabelEncode: # Class handling label
      - DetResizeForTest:
          image_shape: [736, 1280]
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: ['image', 'shape', 'polys', 'ignore_tags']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1 # must be 1
    num_workers: 8
    use_shared_memory: False

**Comment by LDOUBLEV** [comment]
你这是还没开始训练，启动训练后看一下你的内存和显存占用，程序是否在正常运行

**Comment by Xu-xunshan** [comment]
我也是一样的情况内存利用率挺高的，但是只要显卡利用率一动他就结束程序

**Comment by Xu-xunshan** [comment]
我也是一样的情况内存利用率挺高的，但是只要显卡利用率一动他就结束程序

**Comment by Sest0** [comment]
> 你这是还没开始训练，启动训练后看一下你的内存和显存占用，程序是否在正常运行

您好，经过调整训练正常了，但是跑1个epoch就会停住，没法保存模型
![T09CWX5Q`9 PRQOR522ONRW](https://user-images.githubusercontent.com/64736661/111103987-9a8f5d80-858a-11eb-80be-46b2650fa6c1.png)

**Comment by Sest0** [comment]
> 我也是一样的情况内存利用率挺高的，但是只要显卡利用率一动他就结束程序

配置文件里batch_size调小些或者调成1试试

**Comment by Xu-xunshan** [comment]
> > 我也是一样的情况内存利用率挺高的，但是只要显卡利用率一动他就结束程序
> 
> 配置文件里batch_size调小些或者调成1试试
一样我这能跑了但是也跑不完一个epoch

**Comment by SWHL** [comment]
我也遇到同样的问题

**Comment by jimyho-msn** [comment]
我也遇到这个问题了，Windows 10,  PaddlePaddle 2.0.0,  2.0.1  动态版，cpu/gpu 版本都试过了。 
没有报错，自然退出，也没有模型生成。
![Uploading Train_11.png…]()

**Comment by jimyho-msn** [comment]
![Train_11](https://user-images.githubusercontent.com/66407841/112438931-72baa980-8d83-11eb-9f79-3e126c9e0851.png)

**Comment by SWHL** [comment]
- @jimyho-msn 可以去看一下生成的日志，大概是`log/workerlog.0`,这个下面的输出应该会有报错信息，终端有部分信息不能输出
- 我前几天也总是遇见这个问题，但是根据`log/workerlog.0`中总是`DataLoader`的报错信息，于是怀疑图像可能有部分标注不合理导致的，主要是框坐标点出现负值或者超出图像宽高
- 清洗后，目前仍在训练中
- 希望可以帮助到大家

**Comment by jimyho-msn** [comment]
@SWHL，谢谢你的回答，
请问 log/workerlog.0 会在什么地方？ 我怎么用Everything 都找不到。。

**Comment by SWHL** [comment]
我用的是release/2.0 分支下的，`train.sh`如下，日志存放位置`--log_dir=./debug/`，`workerlog.0`会在`train.sh`所在`debug`目录下
https://github.com/PaddlePaddle/PaddleOCR/blob/d3a4f788646efd4a3084f5a9807b679177fadfcd/train.sh#L2

**Comment by Xu-xunshan** [comment]
> * @jimyho-msn 可以去看一下生成的日志，大概是`log/workerlog.0`,这个下面的输出应该会有报错信息，终端有部分信息不能输出
> * 我前几天也总是遇见这个问题，但是根据`log/workerlog.0`中总是`DataLoader`的报错信息，于是怀疑图像可能有部分标注不合理导致的，主要是框坐标点出现负值或者超出图像宽高
> * 清洗后，目前仍在训练中
> * 希望可以帮助到大家

我的应该不是数据的问题，因为我拿同一份数据换了Linux就能训练，Windows底下就不行很迷惑

**Comment by jimyho-msn** [comment]
我用CentOS7 虚拟机可以跑。但不能用gpu, 速度非常慢。
在windows 10 就老不能生成模型。。
